src/python/pants/option/options_integration_test.py

```diff
@@ -1,315 +1,96 @@
-
-        1. An invalid global option `invalid_global` will be caught.
-        2. Variable `invalid_global` is not allowed in [GLOBAL].
+        def rules():
+            return [SubsystemRule(Options)]
         """
-        with temporary_dir(root_dir=os.path.abspath(".")) as tempdir:
-            config_path = os.path.relpath(os.path.join(tempdir, "config.toml"))
-            with open(config_path, "w+") as f:
-                f.write(
-                    dedent(
-                        """
-                        [DEFAULT]
-                        some_crazy_thing = 123
-
-                        [GLOBAL]
-                        invalid_global = true
-                        another_invalid_global = false
-                        """
-                    )
-                )
-            pants_run = self.run_pants([f"--pants-config-files={config_path}", "goals"])
-            pants_run.assert_failure()
-            self.assertIn("ERROR] Invalid option 'invalid_global' under [GLOBAL]", pants_run.stderr)
-            self.assertIn(
-                "ERROR] Invalid option 'another_invalid_global' under [GLOBAL]",
-                pants_run.stderr,
-            )
-
-    def test_invalid_command_line_option_and_invalid_config(self) -> None:
-        """Make sure invalid command line error will be thrown and exits."""
-        with temporary_dir(root_dir=os.path.abspath(".")) as tempdir:
-            config_path = os.path.relpath(os.path.join(tempdir, "config.toml"))
-            with open(config_path, "w+") as f:
-                f.write(
-                    dedent(
-                        """
-                        [pytest]
-                        bad_option = true
-
-                        [invalid_scope]
-                        abc = 123
-                        """
-                    )
-                )
-
-            # Run with invalid config and invalid command line option.
-            # Should error out with invalid command line option only.
-            pants_run = self.run_pants(
-                [f"--pants-config-files={config_path}", "--pytest-invalid=ALL", "goals"]
-            )
-            pants_run.assert_failure()
-            self.assertIn(
-                "Unrecognized command line flag '--invalid' on scope 'pytest'",
-                pants_run.stderr,
-            )
-
-            # Run with invalid config only.
-            # Should error out with `bad_option` and `invalid_scope` in config.
-            pants_run = self.run_pants([f"--pants-config-files={config_path}", "goals"])
-            pants_run.assert_failure()
-            self.assertIn("ERROR] Invalid option 'bad_option' under [pytest]", pants_run.stderr)
-            self.assertIn("ERROR] Invalid scope [invalid_scope]", pants_run.stderr)
-
-    def test_command_line_option_unused_by_goals(self) -> None:
-        self.run_pants(["filter", "--test-debug"]).assert_success()
-        self.run_pants(["filter", "--debug"]).assert_failure()
```

Comment:
Seems unnecessary to have a costly IT. A goal option is very similar to any other subsystem now.